runtime/rastrace/method_trace.c

```diff
@@ -477,10 +480,28 @@ traceMethodArgObject(J9VMThread *thr, UDATA* arg0EA, char* cursor, UDATA length)
 		J9Class* clazz = J9OBJECT_CLAZZ(thr, object);
 		J9ROMClass * romClass = clazz->romClass;
 		J9UTF8* className = J9ROMCLASS_CLASSNAME(romClass);
+		J9JavaVM *vm = thr->javaVM;
+
+		if (clazz == J9VMJAVALANGSTRING_OR_NULL(vm)) {
+			/* string arg */
+			char stringArgBuffer[DEFAULT_BUFFER_LENGTH];
 
-		/* TODO: handle arrays */
+			J9InternalVMFunctions const * const vmFuncs = thr->javaVM->internalVMFunctions;
+			char *stringArgUTF8 = vmFuncs->copyStringToUTF8WithMemAlloc(thr, object, J9_STR_NULL_TERMINATE_RESULT, "  ", 2, stringArgBuffer, DEFAULT_BUFFER_LENGTH, NULL);
+
+			if(DEFAULT_STRING_LENGTH < strlen(stringArgUTF8)) {
+				j9str_printf(PORTLIB, cursor, length, "(String)%.*s...", (U_32)DEFAULT_STRING_LENGTH, J9UTF8_DATA(stringArgUTF8));
+			} else {
+				j9str_printf(PORTLIB, cursor, length, "(String)%.*s", (U_32)J9UTF8_LENGTH(stringArgUTF8), J9UTF8_DATA(stringArgUTF8));
+			}
```

Comment:
This needs to handle the possible failure of `copyStringToUTF8WithMemAlloc()` and we can now use `utf8Length`.
I think the strings should be quoted to distinguish from the failure case.
```c
			if (NULL == utf8String) {
				j9str_printf(PORTLIB, cursor, length, "(String)<error>");
			} else if (utf8Length > DEFAULT_STRING_LENGTH) {
				j9str_printf(PORTLIB, cursor, length, "(String)\"%.*s\"...", (U_32)DEFAULT_STRING_LENGTH, utf8String);
			} else {
				j9str_printf(PORTLIB, cursor, length, "(String)\"%.*s\"", (U_32)utf8Length, utf8String);
			}
```